src/main/java/com/nvidia/spark/rapids/jni/CastStrings.java

```diff
@@ -156,6 +159,163 @@ public static ColumnVector fromIntegersWithBase(ColumnView cv, int base) {
+   * org/apache/spark/sql/catalyst/util/SparkDateTimeUtils.scala#L544
+   *
+   * Use the default timezone if timestamp string does not contain timezone.
+   *
+   * Supports the following formats:
+   * `[+-]yyyy[y][y]`
+   * `[+-]yyyy[y][y]-m[m]`
+   * `[+-]yyyy[y][y]-m[m]-d[d]`
+   * `[+-]yyyy[y][y]-m[m]-d[d] `
+   * `[+-]yyyy[y][y]-m[m]-d[d] [h]h:[m]m:[s]s.[ms][ms][ms][us][us][us][zone_id]`
+   * `[+-]yyyy[y][y]-m[m]-d[d]T[h]h:[m]m:[s]s.[ms][ms][ms][us][us][us][zone_id]`
+   *
+   * The max length of yyyy[y][y] is 6 digits.
+   *
+   * Supports the following zone id forms:
+   * - Z - Zulu timezone UTC+0
+   * - +|-[h]h:[m]m
+   * - A short id, see
+   * https://docs.oracle.com/javase/8/docs/api/java/time/ZoneId.html#SHORT_IDS
+   * - An id with one of the prefixes UTC+, UTC-, GMT+, GMT-, UT+ or UT-,
+   * and a suffix in the formats:
+   * - +|-h[h]
+   * - +|-hh[:]mm
+   * - +|-hh:mm:ss
+   * - +|-hhmmss
+   * - Region-based zone IDs in the form `area/city`, such as `Europe/Paris`
+   *
+   * @param input The input String column contains timestamp strings
+   * @param defaultTimeZone The default timezone to be used. If there is no timezone in string, 
+   *   use this default timezone
+   * @param ansi_enabled Throw exception if has any invalid data and ansi_enabled is true,
+   * otherwise return null
+   * @return a timestamp column or null if has any invalid data and ansi_enabled is true
+   */
+  public static ColumnVector toTimestamp(
+      ColumnView input,
+      String defaultTimeZone,
+      boolean ansi_enabled) {
+
+    // 1. check default timezone is valid
+    Integer defaultTimeZoneIndex = GpuTimeZoneDB.getIndexToTransitionTable(defaultTimeZone);
+    if (defaultTimeZoneIndex == null) {
+      throw new IllegalArgumentException("Invalid default timezone: " + defaultTimeZone);
+    }
+
+    long defaultEpochDay = LocalDate.now(ZoneId.of(defaultTimeZone)).toEpochDay();
+
+    // 2. parse to intermediate result
+    try (ColumnVector tzInfo = GpuTimeZoneDB.getTimeZoneInfo();
+        ColumnVector parseResult = parseTimestampStrings(
+            input, defaultTimeZoneIndex, defaultEpochDay, tzInfo)) {
+
+      ColumnView invalid = parseResult.getChildColumnView(0);
+      ColumnView tsSeconds = parseResult.getChildColumnView(1); // seconds col
+      ColumnView tsMicroseconds = parseResult.getChildColumnView(2);
+      ColumnView tzType = parseResult.getChildColumnView(3);
+      ColumnView tzOffset = parseResult.getChildColumnView(4);
+      ColumnView tzIndex = parseResult.getChildColumnView(6);
+
+      // 3. fallback to cup if has any DST timezone and has any timestamp exceeds max
+      // year
+      ColumnView hasDSTCv = parseResult.getChildColumnView(5); // DST col
+      boolean exceedsMaxYearThresholdOfDST = GpuTimeZoneDB.exceedsMaxYearThresholdOfDST(tsSeconds);
+      boolean hasDST;
+      try (Scalar s = hasDSTCv.sum(DType.INT32)) {
+        hasDST = s.isValid() && s.getInt() > 0;
+      }
```

Comment:
```suggestion
      boolean hasDST;
      try (Scalar s = hasDSTCv.any()) {
        hasDST = s.isValid() && s.getBoolean();
      }
```

Reply:
I remember Nghia said that cuDF logical operator is slow than math operator. So here use sum instead of any.